server/sonar-server/src/test/java/org/sonar/server/notification/ws/AddActionTest.java

```diff
@@ -144,6 +147,7 @@ public void add_a_project_notification_when_a_global_one_exists() {
     ComponentDto project = db.components().insertPrivateProject();
```

Comment:
a variant of this test should exist with a public project

Reply:
same goes with test "fail_when_unknown_project_dispatcher", there should be a variant with a public project.
Alternatively for "fail_when_unknown_project_dispatcher", the test could be written with:
```
ComponentDto project = new Random().nextBoolean() ? db.components().insertPrivateProject() : db.components().insertPublicProject();
```

Reply:
I've created two tests since it's very fast because using the test with Random() with an issue will lead to working 1/2 and will may be treated as a FalsePositive.